nhd/NHDScheduler.py

```diff
@@ -58,44 +58,45 @@ def __init__(self, q: Queue):
         self.logger.warning(f'NHD version {self.ver}')
 
 
-    def InitNHDNodes(self):
+    def InitNHDNodes(self,newNodes):
         """
         Find all nodes handled by NHD given the node taints
         """        
-        nodes = self.k8s.GetNodes()
-        for node in nodes:
+        for node in newNodes:
             active = self.k8s.IsNodeActive(node)
             self.nodes[node] = Node(node, active)
 
         schedulable = sum([node.active for node in self.nodes.values()])
 
         self.logger.info(f'{schedulable} nodes are marked as schedulable by NHD out of {len(self.nodes)} total nodes: {self.nodes.keys()}')
 
-    def BuildInitialNodeList(self):
+    def BuildInitialNodeList(self,newNodes):
         """
         Builds the initial list of available nodes on startup.
         In the future we need to watch for downed nodes and new nodes added later.
         """
         self.logger.info("Populating list of nodes") 
-        self.InitNHDNodes()
+        self.InitNHDNodes(newNodes)
+
 
         for n, v in self.nodes.items():
-            try:
-                v.SetNodeAddr(self.k8s.GetNodeAddr(n))
+            if n in newNodes:
+                try:
+                    v.SetNodeAddr(self.k8s.GetNodeAddr(n))
 
-                if not v.ParseLabels(self.k8s.GetNodeLabels(n)):
-                    self.logger.error(f'Error while parsing labels for node {n}, deactivating node')
-                    v.active = False
-                    continue
+                    if not v.ParseLabels(self.k8s.GetNodeLabels(n)):
+                        self.logger.error(f'Error while parsing labels for node {n}, deactivating node')
+                        v.active = False
+                        continue
 
-                (alloc, free) = self.k8s.GetNodeHugepageResources(n) 
-                if alloc == 0 or not v.SetHugepages(alloc, free):
-                    self.logger.error(f'Error while parsing allocatable resources for node {n}, deactivating node')
-                    v.active = False                  
+                    (alloc, free) = self.k8s.GetNodeHugepageResources(n) 
+                    if alloc == 0 or not v.SetHugepages(alloc, free):
+                        self.logger.error(f'Error while parsing allocatable resources for node {n}, deactivating node')
+                        v.active = False                  
 
-            except Exception as e:
-                self.logger.error(f'Caught exception while setting up node {n}:\n    {e}')
-                v.active = False
+                except Exception as e:
+                    print('Caught exception while setting up node {n}:', e)
```

Comment:
Please change this message back to the previous `self.logger.error()` message.

Reply:
Solved